client/ayon_core/hosts/substancepainter/api/lib.py

```diff
@@ -640,3 +679,88 @@ def _setup_prompt():
         return
 
     return project_mesh
+
+
+def get_export_presets_by_filtering(export_preset_name, channel_type_names):
+    """Function to get export presets included with specific channels
+    requested by users.
+
+    Args:
+        export_preset_name (str): Name of export preset
+        channel_type_list (list): A list of channel type requested by users
+
+    Returns:
+        dict: export preset data
+    """
+
+    target_maps = []
+
+    export_presets = get_export_presets()
+    export_preset_nice_name = export_presets[export_preset_name]
+    resource_presets = substance_painter.export.list_resource_export_presets()
+    preset = next(
+        (
+            preset for preset in resource_presets
+            if preset.resource_id.name == export_preset_nice_name
+        ), None
+    )
+    if preset is None:
+        return {}
+
+    maps = preset.list_output_maps()
+    for channel_map in maps:
+        for channel_name in channel_type_names:
+            if not channel_map.get("fileName"):
+                continue
+
+            if channel_name in channel_map["fileName"]:
+                target_maps.append(channel_map)
+    # Create a new preset
+    return {
+        "exportPresets": [
+            {
+                "name": export_preset_name,
+                "maps": target_maps
+            }
+        ],
+    }
+
+
+@contextlib.contextmanager
+def set_layer_stack_opacity(node_ids, channel_types):
+    """Function to set the opacity of the layer stack during
+    context
+    Args:
+        node_ids (list): A list of substance painter node ids
+        channel_types (list): A list of channel types
+    """
+    # Do nothing
+    if not node_ids or not channel_types:
+        yield
+        return
+
+    stack = substance_painter.textureset.get_active_stack()
+    stack_root_layers = (
+        substance_painter.layerstack.get_root_layer_nodes(stack)
+    )
+    all_selected_nodes = []
+    for node_id in node_ids:
+        node = substance_painter.layerstack.get_node_by_uid(int(node_id))
+        all_selected_nodes.append(node)
+    excluded_nodes  = {node for node in stack_root_layers
+                       if node not in all_selected_nodes}
```

Comment:
This would be nicer if we could get this in one go instead of two for loops.
```python
    excluded_nodes = []
    for root_layer_node in stack_root_layers:
        node_id = root_layer_node.uid()    # not sure how to get the id from a node?
        if node_id not in node_ids:
            excluded_nodes.append(root_layer_node)
```
Or even:
```python
    excluded_nodes = [
        node for node in stack_root_layers
        if node.uid() not in node_ids
    ]
```

If the node `uid` is also an `int` then we can benefit from the optimization of first doing `node_ids = set(node_ids)` so that the lookup os `O1` instead of `On` 

Making this:

```suggestion
    node_ids = set(node_ids)  # lookup
    excluded_nodes = [
        node for node in stack_root_layers
        if node.uid() not in node_ids
    ]
```